aea/protocols/base.py

```diff
@@ -340,15 +340,25 @@ def from_config(cls, configuration: ProtocolConfig) -> "Protocol":
             configuration.prefix_import_path + ".message"
         )
         classes = inspect.getmembers(class_module, inspect.isclass)
-        message_classes = list(filter(lambda x: re.match("\\w+Message", x[0]), classes))
+        name_camel_case = "".join(
+            word.capitalize() for word in configuration.name.split("_")
+        )
+        message_classes = list(
+            filter(
+                lambda x: re.match("{}Message".format(name_camel_case), x[0]), classes
+            )
+        )
         assert len(message_classes) == 1, "Not exactly one message class detected."
         message_class = message_classes[0][1]
         class_module = importlib.import_module(
             configuration.prefix_import_path + ".serialization"
         )
         classes = inspect.getmembers(class_module, inspect.isclass)
         serializer_classes = list(
-            filter(lambda x: re.match("\\w+Serializer", x[0]), classes)
+            filter(
+                lambda x: re.match("{}Serializer".format(name_camel_case), x[0]),
+                classes,
+            )
```

Comment:
@marcofavorito since we generate protocols and have a convention here on naming I now enforced this. It was needed as the previous implementations clashed with the signing protocol where we also have `RawMessage` custom type which then clashes with this mechanism.

Reply:
ok